### DISCUSSION TITLE
How does Argo store the workflows?

### DISCUSSION THREAD

**Discussion by nipunbharti21** [argoproj/argo-workflows#8536] [Q&A] [answered]
I am using Argo v2.7.6 and recently updated it to v3.0. But argo cli stopped listing workflows when I run `argo list workflows -A`. Even calling the API `/api/v1/workflows/argo` does not list any workflows. I checked the `argo_archived_workflows` database which does contain all the workflows but Argo doesn't seem to be able to list them. How does Argo fetch the workflows and where are they actually stored for argo to access them?

**Comment by terrytangyuan** [comment, marked as answer]
They are stored in Kubernetes etcd. It's likely that you haven't updated your Argo CLI to v3.0. Could you check? Did you get anything from `kubectl get wf`?
